h2/src/main/org/h2/command/dml/ExecuteImmediate.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2004-2019 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (https://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.command.dml;
+
+import org.h2.api.ErrorCode;
+import org.h2.command.CommandInterface;
+import org.h2.command.Prepared;
+import org.h2.engine.Session;
+import org.h2.expression.Expression;
+import org.h2.message.DbException;
+import org.h2.result.ResultInterface;
+
+/**
+ * This class represents the statement
+ * EXECUTE IMMEDIATE.
+ */
+public class ExecuteImmediate extends Prepared {
+
+    private Expression statement;
+
+    public ExecuteImmediate(Session session, Expression statement) {
+        super(session);
+        this.statement = statement.optimize(session);
+    }
+
+    @Override
+    public int update() {
+        String sql = statement.getValue(session).getString();
+        if (sql == null) {
+            throw DbException.getInvalidValueException("SQL command", null);
+        }
+        Prepared command = session.prepare(sql);
+        if (command.isQuery()) {
+            throw DbException.get(ErrorCode.SYNTAX_ERROR_2, sql, "<not a query>");
```

Comment:
not a query -> may not be a query

Reply:
The actual message is
```
Syntax error in SQL statement "SELECT 1"; expected "<not a query>"; SQL statement:
EXECUTE IMMEDIATE 'SELECT 1' [42001-199]
```
Maybe I'm wrong, but from my point of view “may not be a query” does not fit into that context.

Reply:
You are correct